### PR TITLE
Try to fix dump_local_console (fixes #5488)

### LINKED ISSUE #5488: dump_local_console seems not to be correct UTF-8
As reported on Discord by Ham5terzilla:

> russian lang in local console dump is corrupted, example:
> 
> [2022-06-24 12:30:08][chat]: NT RYSKI: Админ ебаный хуй
> ->
> [2022-06-24 12:30:08][chat]: NT RYSKI: РђРґРјРёРЅ РµР±Р°РЅС‹Р№ С…СѓР№

### PR DESCRIPTION
**Pull request by def-**
Untested, but my hunch is that by not cutting off the last part the text
file will be correctly recognized as valid utf8

## Checklist

- [ ] Tested the change ingame
- [ ] Provided screenshots if it is a visual change
- [ ] Tested in combination with possibly related configuration options
- [ ] Written a unit test if it works standalone, system.c especially
- [ ] Considered possible null pointers and out of bounds array indexing
- [ ] Changed no physics that affect existing maps
- [ ] Tested the change with [ASan+UBSan or valgrind's memcheck](https://github.com/ddnet/ddnet/#using-addresssanitizer--undefinedbehavioursanitizer-or-valgrinds-memcheck) (optional)

### CODE REVIEW

File: src/game/client/components/console.cpp [hunk 1]
```diff
@@ -285,7 +285,7 @@ void CGameConsole::CInstance::OnInput(IInput::CEvent Event)
 				if(m_Type == CONSOLETYPE_LOCAL || m_pGameConsole->Client()->RconAuthed())
 				{
 					char *pEntry = m_History.Allocate(m_Input.GetLength() + 1);
-					mem_copy(pEntry, m_Input.GetString(), m_Input.GetLength() + 1);
+					str_copy(pEntry, m_Input.GetString(), m_Input.GetLength());
```
Comment (Robyt3) [inline src/game/client/components/console.cpp:288]:
```suggestion
					str_copy(pEntry, m_Input.GetString(), m_Input.GetLength() + 1);
```

Should be the size of the destination buffer, which is what we allocate above.

File: src/game/client/components/console.cpp [hunk 2]
```diff
@@ -425,7 +425,7 @@ void CGameConsole::CInstance::PrintLine(const char *pLine, int Len, ColorRGBA Pr
 	CBacklogEntry *pEntry = m_Backlog.Allocate(sizeof(CBacklogEntry) + Len);
 	pEntry->m_YOffset = -1.0f;
 	pEntry->m_PrintColor = PrintColor;
-	mem_copy(pEntry->m_aText, pLine, Len);
+	str_copy(pEntry->m_aText, pLine, Len);
 	pEntry->m_aText[Len] = 0;
```
Comment (Robyt3) [inline src/game/client/components/console.cpp:429]:
```suggestion
	str_copy(pEntry->m_aText, pLine, Len + 1);
```

**Comment by heinrich5991** [comment]
bors r+

Thanks.

**Comment by heinrich5991** [comment]
The integration test seems flaky…

bors retry

**Pull request merged**